Make `[android] icon` config option optional

Android Emulator build fails if `[android] icon` is missing or have wrong path

```
> Task :app:mergeDevReleaseResources FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeDevReleaseResources'.
> Multiple task action failures occurred:
> A failure occurred while executing com.android.build.gradle.internal.res.Aapt2CompileRunnable
> Android resource compilation failed
        ERROR:/Users/chicoxyzzy/dev/other/qwejs/build/android-emulator/app/src/main/res/mipmap/icon.png: AAPT: error: file failed to compile.

> A failure occurred while executing com.android.build.gradle.internal.res.Aapt2CompileRunnable
> Android resource compilation failed
        ERROR:/Users/chicoxyzzy/dev/other/qwejs/build/android-emulator/app/src/main/res/mipmap/ic_launcher_round.png: AAPT: error: file failed to compile.

> A failure occurred while executing com.android.build.gradle.internal.res.Aapt2CompileRunnable
> Android resource compilation failed
        ERROR:/Users/chicoxyzzy/dev/other/qwejs/build/android-emulator/app/src/main/res/mipmap/ic_launcher.png: AAPT: error: file failed to compile.


* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
82 actionable tasks: 28 executed, 54 up-to-date
• ERROR: failed to invoke `gradlew assemble` command +58997ms
```

--- a/src/cli/cli.cc
+++ b/src/cli/cli.cc
@@ -3373,14 +3373,14 @@ int main (const int argc, const char* argv[]) {
       auto androidIcon = settings["android_icon"];
 
       if (androidIcon.size() > 0) {
+        settings["android_application_icon_config"] = (
+          String("    android:roundIcon=\"@mipmap/ic_launcher_round\"\n") +
+          String("    android:icon=\"@mipmap/ic_launcher\"\n")
+        );
+
         fs::copy(targetPath / androidIcon, res / "mipmap" / "icon.png", fs::copy_options::overwrite_existing);
         fs::copy(targetPath / androidIcon, res / "mipmap" / "ic_launcher.png", fs::copy_options::overwrite_existing);
         fs::copy(targetPath / androidIcon, res / "mipmap" / "ic_launcher_round.png", fs::copy_options::overwrite_existing);
-      } else {
-        // create empty icons
-        std::ofstream icon((res / "mipmap" / "icon.png").string());
-        std::ofstream ic_launcher((res / "mipmap" / "ic_launcher.png").string());
-        std::ofstream ic_launcher_round((res / "mipmap" / "ic_launcher_round.png").string());
       }
 
       // allow user space to override all `res/` files

--- a/src/cli/templates.hh
+++ b/src/cli/templates.hh
@@ -469,8 +469,7 @@ constexpr auto gAndroidManifest = R"XML(
     android:label="{{meta_title}}"
     android:theme="@style/Theme.AppCompat.Light"
     android:supportsRtl="true"
-    android:roundIcon="@mipmap/ic_launcher_round"
-    android:icon="@mipmap/ic_launcher"
+    {{android_application_icon_config}}
     {{android_allow_cleartext}}
   >
     <activity

